scalding-parquet-scrooge/src/main/java/com/twitter/scalding/parquet/scrooge/ScroogeReadSupport.java

```diff
@@ -185,4 +187,54 @@ public static void assertAreCompatible(Type fileType, Type projection) {
     }
   }
 
+  /**
+   * Overriding to fall back to get descriptor from the {@link #thriftClass} if thrift metadata is
+   * not present
+   *
+   * @return
+   */
+  @Override
+  public RecordMaterializer<T> prepareForRead(Configuration configuration,
+                                              Map<String, String> keyValueMetaData, MessageType fileSchema,
+                                              ReadSupport.ReadContext readContext) {
+    ThriftMetaData thriftMetaData = ThriftMetaData.fromExtraMetaData(keyValueMetaData);
+    try {
+      if (thriftClass == null) {
+        thriftClass = getThriftClass(keyValueMetaData, configuration);
+      }
+
+      ThriftType.StructType descriptor = null;
+      if (thriftMetaData != null) {
+        descriptor = thriftMetaData.getDescriptor();
+      } else {
+        ScroogeStructConverter schemaConverter = new ScroogeStructConverter();
+        descriptor = schemaConverter.convert(thriftClass);
+      }
+
+      ThriftRecordConverter<T> converter = new ScroogeRecordConverter<T>(
```

Comment:
Is it okay to assume `ScroogeRecordConverter` instead of reading from the config like https://github.com/apache/parquet-mr/blob/master/parquet-thrift/src/main/java/org/apache/parquet/hadoop/thrift/ThriftReadSupport.java#L244?

Reply:
paging @isnotinvain and @julienledem any comments here?

Having more flexibility would help our interop with spark which does not write this metadata by default.

Reply:
Ack. Thanks for confirming. If it helps, I can follow up with the same changes there.